### PR TITLE
`Programming exercises`: Improve styling of feedback and manual assessment

### PR DESCRIPTION
**Pull request by pzdr7**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

#### Changes affecting Programming Exercises
- [x] **High priority**: I tested **all** changes and their related features with **all** corresponding user types on a test server configured with the **integrated lifecycle setup** (LocalVC and LocalCI).

### Motivation and Context

There are a few visual issues in the manual assessment of programming exercises and with feedback widgets in general:

- Poor contrast between build annotations and the highlighted lines (mentioned in #8437)
- Build annotations "jumping" to the right when the assessment button is on the same line (#8731)
- Feedback widgets wasting a large amount of space for long text (#8730)

### Description

- Highlighted lines in the assessment view are now only highlighted in the line part; the margin is left for the build annotations
- The button to add feedback is now to the right of the line numbers (the line decorations), ensuring that it does not interfere with the line numbers (at the cost of some testability & flexibility). This replaces the old glyph margin hover button.
- Feedbacks are now allowed to be much wider. They render more text in fewer lines (given that there are no line breaks)
- Made the resize handles more prominent in light mode

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming exercise with
  - SCA enabled
  - Manual assessment enabled
  - 1 submission that has at least one SCA warning (if you enable all SCA categories, you can trigger a warning by forgetting whitespace after `if`, e.g. `if(input.size() > 0)`)

1. Log in to Artemis
2. Go to your course > exercises > your programming exercise > scores
3. Open the assessment of the submission
4. Go to the file with the SCA warning and verify that the button does not clash with the build annotation (⚠️ icon)
5. Verify the changed lines are highlighted and that everything looks okay in light and dark mode
6. Add a long feedback without newlines
7. Verify that the feedback widget does not waste space when displaying the long feedback text

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| monaco-editor-line-decorations-hover-button.model.ts | 91.17% | ✅ |
| monaco-editor.component.ts | 97.87% | ✅ |

#### Feedback widgets

<details>
  <summary>
    Before
  </summary>

image
</details>

<details>
  <summary>
    After
  </summary>

image
</details>

#### Feedback button / line highlights / build annotations

<details>
  <summary>
    Before
  </summary>

image
</details>

<details>
  <summary>
    After
  </summary>

image
</details>

#### Drag handles in the code editor grid

<details>
  <summary>
    Before
  </summary>

image
</details>

<details>
  <summary>
    After
  </summary>

image
</details>

## Summary by CodeRabbit

- **New Features**
  - Introduced a new hover button for line decorations in the code editor.

- **Bug Fixes**
  - Removed an unused feedback button from the code editor.

- **Refactor**
  - Replaced `MonacoEditorGlyphMarginHoverButton` with `MonacoEditorLineDecorationsHoverButton` for improved interaction handling.

- **Tests**
  - Updated test cases to reflect changes in hover button functionality and ensure accurate line highlighting.

### CODE REVIEW

File: src/main/webapp/app/shared/monaco-editor/monaco-editor.component.ts
Comment (pzdr7) [inline src/main/webapp/app/shared/monaco-editor/monaco-editor.component.ts:271]:
```suggestion
            `line-decorations-hover-button-${this._editor.getId()}`,
```

**Review by BaumiCoder — DISMISSED** [review]
I tested it on my local machine. 
1. :heavy_check_mark: less margin and more width for the feedback box and its text
2. :heavy_check_mark: No more flickering through the warning

(I could not test it fully on a test server, because there is no checkbox `Enable Static Code Analysis`.)

**Review by b-fein — APPROVED** [review]
Code changes and screenshots look good.

Thanks!

**Review by JohannesStoehr — DISMISSED** [review]
Code and screenshots look good to me

**Review by Jan-Thurner — DISMISSED** [review]
code looks good

**Review by egekurt123 — APPROVED** [review]
Code LGTM

**Review by JohannesStoehr — APPROVED** [review]
Reapprove

**Review by b-fein — APPROVED** [review]
Re-approve after small merge-conflict.

**Review by Jan-Thurner — APPROVED** [review]
reapprove

**Review by BaumiCoder — APPROVED** [review]
Re-approve after resolved merge conflicts

**Pull request merged**
